autoload/slime.vim

```diff
@@ -122,30 +138,61 @@ endfunction
 """""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
 
 function! slime#send(text)
-  call s:SlimeGetConfig()
-
-  " this used to return a string, but some receivers (coffee-script)
-  " will flush the rest of the buffer given a special sequence (ctrl-v)
-  " so we, possibly, send many strings -- but probably just one
-  let pieces = s:_EscapeText(a:text)
-  for piece in pieces
-    if type(piece) == 0  " a number
-      if piece > 0  " sleep accepts only positive count
-        execute 'sleep' piece . 'm'
+  if s:SlimeDispatchValidate("ValidEnv")
+    try
+      call s:SlimeGetConfig()
+    catch \invalid config\
+      return
+    endtry
+
+    " this used to return a string, but some receivers (coffee-script)
+    " will flush the rest of the buffer given a special sequence (ctrl-v)
+    " so we, possibly, send many strings -- but probably just one
+    let pieces = s:_EscapeText(a:text)
+    for piece in pieces
+      if type(piece) == 0  " a number
+        if piece > 0  " sleep accepts only positive count
+          execute 'sleep' piece . 'm'
+        endif
+      else
+        call s:SlimeDispatch('send', b:slime_config, piece)
       endif
-    else
-      call s:SlimeDispatch('send', b:slime_config, piece)
-    endif
-  endfor
+    endfor
+  endif
 endfunction
 
 function! slime#config() abort
   call inputsave()
-  call s:SlimeDispatch('config')
+  if s:SlimeDispatchValidate("ValidEnv")
+    call s:SlimeDispatch('config')
+
+    if exists("b:slime_config") && !s:SlimeDispatchValidate("ValidConfig", b:slime_config, 0)
+        unlet b:slime_config
+    endif
+  endif
   call inputrestore()
 endfunction
 
 " delegation
+function! s:SlimeDispatchValidate(name, ...)
+  " allow custom override
+  let override_fn = "SlimeOverride" . slime#common#capitalize(a:name)
+  if exists("*" . override_fn)
+    return call(override_fn, a:000)
+  endif
+
+  let fun_string = "slime#targets#" . slime#config#resolve("target") . "#" . a:name
+  " using try catch because exists() doesn't detect autoload functions that aren't yet loaded
+  " the idea is to return the interger 1 for true in cases where a target doesn't have
+  " the called validation function implemented. E117 is 'Unknown function'.
+  try
+    return call(fun_string, a:000)
+  catch /^Vim\%((\a\+)\)\=:E117:/
+    return 1
+  endtry
+
+endfunction
```

Comment:
I'm warming up to this ^

A few ideas:
- what if we remove line 178-184, and delegate to `s:SlimeDispatch(name, ...)` below. It seems most of this logic is duplicated, no?
- we keep the try-catch wrap
- maybe rename to `s:SlimeDispatchSafe`? (I'm open to a better name)

```vim
# caller
if exists("b:slime_config") && !s:SlimeDispatchSafe("ValidateConfig", b:slime_config, 0)
  ...
```

```vim
# (untested) idea
function! s:SlimeDispatchSafe(name, ...)
  " using try catch because exists() doesn't detect autoload functions that aren't yet loaded
  " the idea is to return the interger 1 for true in cases where a target doesn't have
  " the called validation function implemented. E117 is 'Unknown function'.
  try
    return s:SlimeDispatch(name, a:000)
  catch /^Vim\%((\a\+)\)\=:E117:/
    return 1
  endtry
endfunction
```

Reply:
Hi! Sorry for disappearing for a few months.   I essentially took your suggestion and now have:

```vim
function! s:SlimeDispatchValidate(name, ...)
  " using try catch because exists() doesn't detect autoload functions that aren't yet loaded
  " the idea is to return the interger 1 for true in cases where a target doesn't have
  " the called validation function implemented. E117 is 'Unknown function'.
  try
    return call ("s:SlimeDispatch", [a:name] + a:000)
  catch /^Vim\%((\a\+)\)\=:E117:/
    return 1
  endtry
endfunction

function! s:SlimeDispatch(name, ...)
  " allow custom override
  let override_fn = "SlimeOverride" . slime#common#capitalize(a:name)
  if exists("*" . override_fn)
    return call(override_fn, a:000)
  endif
  return call("slime#targets#" . slime#config#resolve("target") . "#" . a:name, a:000)
endfunction
```

I still call it `s:SlimeDispatchValidate` because its purpose is still validation (hence the default return value of 1 for functions that don't exist). 

I've manually tested this in on the neovim target, the tmux target from both vim and neovim, and the vimterminal target.